torch/distributed/tensor/_ops/_math_ops.py

```diff
@@ -291,6 +291,13 @@ def common_reduction_strategy(
                     reduction_linear = False
                     break
 
+        for p in op_spec.output_spec.placements:
```

Comment:
I suspect this change is too aggressive, shouldn't it consider 'reduction_dims' and only set reduction_linear to False if the reduce op is happening on a dim that is partial and differs from reduce_op?